src/main/java/seedu/toluist/controller/FindController.java

```diff
@@ -81,24 +84,30 @@ public CommandResult execute(String command) {
 
     private CommandResult formatDisplay(boolean isSearchByTag, boolean isSearchByName,
                                         String[] keywordList, int foundCount) {
-        String searchParameters;
-        if (isSearchByName && isSearchByTag) {
-            searchParameters = NAME_AND_TAG_MESSAGE;
-        } else if (isSearchByName) {
-            searchParameters = NAME_MESSAGE;
-        } else { //isSearchByTag
-            searchParameters = TAG_MESSAGE;
-        }
+        if (keywordList[0].equals(NULL_PARAMETER)) {
+            return new CommandResult(String.format(LIST_RESULT_MESSAGE_TEMPLATE, foundCount));
+        } else {
+            String searchParameters;
+
+            if (isSearchByName && isSearchByTag) {
+                searchParameters = NAME_AND_TAG_MESSAGE;
+            } else if (isSearchByName) {
+                searchParameters = NAME_MESSAGE;
+            } else { //isSearchByTag
+                searchParameters = TAG_MESSAGE;
+            }
 
-        String keywords = String.join(" ", keywordList);
-        return new CommandResult(String.format(RESULT_MESSAGE_TEMPLATE, keywords, searchParameters, foundCount));
+            String keywords = String.join(STRING_JOINING_MESSAGE, keywordList);
+            return new CommandResult(String.format(FIND_RESULT_MESSAGE_TEMPLATE,
+                                     keywords, searchParameters, foundCount));
+        }
     }
 
     public HashMap<String, String> tokenize(String command) {
         HashMap<String, String> tokens = new HashMap<>();
 
         // search by tag
-        if (command.contains(TAG_PARAMETER)) {
+        if (command.contains(TAG_PARAMETER) || !command.contains(NAME_PARAMETER)) {
```

Comment:
the tests still work I think...
don't really know how to read the tests you put in.
At the very least, coverage increased, because now we have tests for all 3 situations ;)